library/windows/scripts/build_jsonlib.bat

```diff
@@ -0,0 +1,116 @@
+:: Copyright 2018 Google LLC
+::
+:: Licensed under the Apache License, Version 2.0 (the "License");
+:: you may not use this file except in compliance with the License.
+:: You may obtain a copy of the License at
+::
+::      http://www.apache.org/licenses/LICENSE-2.0
+::
+:: Unless required by applicable law or agreed to in writing, software
+:: distributed under the License is distributed on an "AS IS" BASIS,
+:: WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+:: See the License for the specific language governing permissions and
+:: limitations under the License.
+@echo off
+
+:: Find where VS lives and start a VC command prompt
+set pre=Microsoft.VisualStudio.Product.
+set ids=%pre%Community %pre%Professional %pre%Enterprise %pre%BuildTools
+ 
+pushd "C:\Program Files (x86)\Microsoft Visual Studio\Installer\"
+for /f "usebackq tokens=1* delims=: " %%i in (`vswhere -latest -products *`) do (if /i "%%i"=="installationPath" set InstallDir=%%j)
+popd
+
+pushd %InstallDir%\VC\Auxiliary\Build
+call vcvarsall.bat x86_amd64
+popd
+
+set JSONHEADERSEXIST=true
+if not exist %~dp0..\dependencies\json\allocator.h set JSONHEADERSEXIST=false
+
+set JSONDEBUGLIBEXISTS=true
+if not exist %~dp0..\dependencies\json\x64\debug\json_vc71_libmtd.lib set JSONDEBUGLIBEXISTS=false
+
+set JSONRELEASELIBEXISTS=true
+if not exist %~dp0..\dependencies\json\x64\release\json_vc71_libmt.lib set JSONRELEASELIBEXISTS=false
+
+if %JSONHEADERSEXIST% == true (
+    if %JSONDEBUGLIBEXISTS% == true (
+        if %JSONRELEASELIBEXISTS% == true (
+            echo jsoncpp found.
+            goto DONE
+        )
+    )
+)
+
+set THIRDPARTYDIREXISTS=true
+if not exist %~dp0..\third_party set THIRDPARTYDIREXISTS=false
+
+if %THIRDPARTYDIREXISTS% == false (
+    mkdir %~dp0..\third_party
+)
+
+set JSONDIREXISTS=true
+if not exist %~dp0..\third_party\jsoncpp set JSONDIREXISTS=false
+
+if %JSONDIREXISTS% == false (
+    mkdir %~dp0..\third_party\jsoncpp
+)
+
+set JSONEXISTS=true
+if not exist %~dp0..\third_party\jsoncpp\README.md set JSONEXISTS=false
+
+:: Clone source
+if %JSONEXISTS% == false (
+    :: PR opened on json cpp for VS2017 support: https://github.com/open-source-parsers/jsoncpp/pull/853
+    echo Cloning via git clone --branch supportvs2017 https://github.com/clarkezone/jsoncpp.git %~dp0..\third_party\jsoncpp
+    call git clone --branch supportvs2017 https://github.com/clarkezone/jsoncpp.git %~dp0..\third_party\jsoncpp
+
+    pushd %~dp0..\third_party\jsoncpp
+
+    call git checkout 3ae7e8073a425c93329c8577a3c813c206322ca4
+
+    popd
+)
+
+:: Copy headers
+if %JSONHEADERSEXIST% == false (
+    echo Copying jsoncpp headers

```

Comment:
Since the source is no longer being removed, the header-copying step can be removed and the VS project can point directly to `third_party\jsoncpp\include`.

That makes for one less step that could cause issues. For instance, if we need to roll jsoncpp, this copy step as written would result in people with existing trees potentially running into hard-to-debug failures due to using stale headers.

(Although I guess come to think of it, the way the script is structured if we roll jsoncpp it wouldn't even trigger a rebuild attempt, which is a problem. We can punt on that until/unless it becomes necessary though, and I'll just file a bug about making the jsoncpp dependency more robust. Let's still eliminate the header copy unless there's a good reason to keep it though.)

Reply:
All above fixed in next update